### PR TITLE
Add in built-in Symbols

### PR DESCRIPTION
**Pull request by Validark**
This implementation is hopefully approaching something more along the lines of how `Symbol` should actually be implemented

### CODE REVIEW

File: include/RuntimeLib.lua [hunk 1]
```diff
@@ -9,43 +9,60 @@ local TS = {}
...
+	Symbol.iterator = Symbol("Symbol.iterator")
+	Symbol.asyncIterator = Symbol("Symbol.asyncIterator")
+	Symbol.match = Symbol("Symbol.match")
+	Symbol.replace = Symbol("Symbol.replace")
+	Symbol.search = Symbol("Symbol.search")
+	Symbol.species = Symbol("Symbol.species")
+	Symbol.split = Symbol("Symbol.split")
+	Symbol.toPrimitive = Symbol("Symbol.toPrimitive")
+	Symbol.toStringTag = Symbol("Symbol.toStringTag")
+	Symbol.unscopables = Symbol("Symbol.unscopables")
```
Comment (evaera) [inline include/RuntimeLib.lua]: I don't think we should include well-known symbols that we are not using. They are not included in the types so there's no reason to have them here.

File: include/RuntimeLib.lua [hunk 2]
```diff
@@ -9,43 +9,60 @@ local TS = {}
 TS.Promise = Promise
 
 local Symbol do
-	Symbol = {}
-	Symbol.__index = Symbol
-	setmetatable(Symbol, {
+	local symbolRegistry = {}
+
+	Symbol = setmetatable({
+		["for"] = function(key)
```
Comment (evaera) [inline include/RuntimeLib.lua]: We must wait for #136 until we can push this

**Comment by evaera** [comment]
This PR is blocked by #136

**Comment by Validark** [comment]
I don't really see the logic of adding them one at a time. Also, I am going to need hasInstance for certain.

I agree we need to resolve the naming issue first.

**Comment by evaera** [comment]
The logic behind adding them as-needed is that we know will not be using all of the well-known symbols. They don't exist in the types so users can't use them anyways. When we add a feature that makes use of a well-known symbol, we can also add it to the runtime library along with adding the type.

**Pull request closed without merging**